packages/http-server/src/__tests__/server.oas.spec.ts

```diff
@@ -29,33 +29,39 @@ async function instantiatePrism(specPath: string) {
     errors: false,
     cors: true,
   });
+
+  const address = await server.listen(port, '127.0.0.1');
+
+  return {
+    close: server.close.bind(server),
+    address,
+  };
 }
 
 describe('GET /pet?__server', () => {
-  let server: IPrismHttpServer;
+  let server: { close: () => Promise<void>; address: string };
```

Comment:
Ok, brace yourself. 

This is going to be a complete waste of your time but I wanted to do this because I'm old and stupid.

TypeScript has a type that infers the returned type from a function — so we can do

`let server: ReturnType<typeof instantiatePrism>` 

…but that will not be enough, because the type is now `Promise<{}>` — while we want what's _INSIDE_ the Promise.

It turns out we can do that using the infer feature in TypeScript 3.8 and the conditional types, so if we write this:

```ts
type ThenArg<T> = T extends Promise<infer U> ? U :
  T extends ((...args: any[]) => Promise<infer U>) ? U :
  T
```

(I think this should handle all the cases, but you never know)

then we can do

```ts
let server: ThenArg<ReturnType<typeof instantiatePrism>>;
```

Reply:
You are crazy.

Reply:
Ok it works, now I need to learn why and how it works.`